base/iterators.jl

```diff
@@ -649,6 +649,51 @@ end
 iterate(it::Drop, state) = iterate(it.xs, state)
 isdone(it::Drop, state) = isdone(it.xs, state)
 
+
+# takewhile
+
+struct TakeWhile{I,P<:Function}
```

Comment:
The predicate doesn't necessarily need to be a function, for example it's not unreasonable to do something like `takewhile(Bool, [1, 1, 0, 1])`.

Reply:
Eh, I would seriously question whether that's a good way to write that... But yes, `Filter` for example does not constrain its function.

Reply:
this is the compress function of python itertools. it exists and seems to be used!
the name is misleading however, i prefer filter(mask) with mask a bitvector.
it is useful, to do bitvector ops. a lot of sat stuff go this way nowaday

Reply:
One will lands to using a unary closure like with find. may be a callable.